ext/user_request.c

```diff
@@ -123,12 +123,17 @@ PHP_FUNCTION(DDTrace_UserRequest_notify_commit)
         }
     }
 
+    if (status < 100 || status > 599) {
+        zend_type_error("Status code must be between 100 and 599");
```

Comment:
Can we have `zend_value_error` here and polyfill it in `compatibility.h` with `#define zend_value_error zend_type_error` - this was added with PHP 8 and existing "misuses" of TypeError replaced by it in php-src.